lib/ansible/cli/__init__.py

```diff
@@ -31,8 +31,12 @@ def check_blocking_io():
         except Exception:
             continue  # not a real file handle, such as during the import sanity test
 
-        if not os.get_blocking(fd):
-            handles.append(getattr(handle, 'name', None) or '#%s' % fd)
+        try:
+            if not os.get_blocking(fd):
+                handles.append(getattr(handle, 'name', None) or '#%s' % fd)
+        except AttributeError:
+            raise SystemExit('ERROR: Ansible requires blocking IO, but this platform '
+                             'does not have os.get_blocking available')
```

Comment:
It's best to only wrap the instruction that might cause an exception. Just one line. Always. Otherwise, we're risking suppressing something else in the future, and it's hard to debug, then.
```suggestion
        try:
            fd_blocking_io_enabled = os.get_blocking(fd)
        except AttributeError:
            raise SystemExit(
                'ERROR: Ansible requires blocking IO, but this platform '
                'does not have os.get_blocking available'
            )

        if not fd_blocking_io_enabled:
            handles.append(getattr(handle, 'name', None) or '#%s' % fd)
```

Reply:
OTOH, Matt is right — we don't care enough about this to sabotage the entire process at this point, I think:
```suggestion
        try:
            fd_blocking_io_enabled = os.get_blocking(fd)
        except AttributeError:
            print(
                'ERROR: Ansible requires blocking IO, but this platform '
                'does not have os.get_blocking available',
                file=sys.stderr,
            )
            fd_blocking_io_enabled = False

        if not fd_blocking_io_enabled:
            handles.append(getattr(handle, 'name', None) or '#%s' % fd)
```

Reply:
Actually, looking closer, a `os_exposes_get_blocking = hasattr(os, 'get_blocking')` could be performed at the very beginning of the function and used as a guard expression to exit the function early, not even going into the loop. So I'd leave the loop contents alone.